maven-model-builder/src/test/java/org/apache/maven/model/building/ComplexActivationTest.java

```diff
@@ -56,5 +56,26 @@ public void testAndConditionInActivation()
         assertEquals( "activated-1", result.getEffectiveModel().getProperties().get( "profile.file" ) );
         assertNull( result.getEffectiveModel().getProperties().get( "profile.miss" ) );
     }
+    
+    public void testConditionExistingAndMissingInActivation() throws Exception
+    {
+        Properties sysProperties = new Properties();
+        sysProperties.setProperty( "myproperty", "test" );
+
+        ModelBuilder builder = new DefaultModelBuilderFactory().newInstance();
+        assertNotNull( builder );
+
+        DefaultModelBuildingRequest request = new DefaultModelBuildingRequest();
+        request.setProcessPlugins( true );
+        request.setPomFile( getPom( "complexEmptyAndMissing" ) );
+        request.setSystemProperties( sysProperties );
+
+        try {
```

Comment:
oh ok!!!

I think it is a great idea. Usually, when I want to test for Exceptions I use annotation approach like having @Test(expected = Exception.class). However, looking at the existing examples of Apache I couldn't find any. So, I actually learnt the Rule approach is kind of equivalent. I am just applying the change.

Thanks.